### PR TITLE
Optimize the addition and replacement logic of `Transform`

### PR DESCRIPTION
**Pull request by cptbtptpbcptdtptp**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Release Notes

- **New Features**
	- Enhanced component dependency management with new functionalities for declaring dependencies.
	- Introduced a new subclass of `Transform` for improved testing scenarios.

- **Bug Fixes**
	- Updated error messages in tests for better clarity and alignment with intended logic.

- **Improvements**
	- Streamlined handling of transforms across various classes, focusing on entity-based management.
	- Improved error handling and resource management in the `MeshRenderer` class.

- **Refactor**
	- Removed redundant properties and methods to enhance code clarity and maintainability across multiple classes.

### CODE REVIEW

File: tests/src/core/Transform.test.ts [hunk 1]
```diff
@@ -59,4 +60,85 @@ describe("Transform test", function () {
...
+    this.size && this.size.copyFrom(state.size);
+  }
+}
+
+interface SubClassOfTransformTransitionState {
+  size: Vector2;
+  position: Vector3;
+  rotation: Vector3;
+  scale: Vector3;
+}
```
Comment (singlecoder) [inline tests/src/core/Transform.test.ts:144]: Not used, should be delete

File: packages/core/src/Entity.ts [hunk 2]
```diff
@@ -193,11 +217,17 @@ export class Entity extends EngineObject {
...
     super(engine);
     this.name = name;
-    this.transform = this.addComponent(Transform);
-    this._inverseWorldMatFlag = this.transform.registerWorldChangeFlag();
+    for (let i = 0, n = components.length; i < n; i++) {
+      const type = components[i];
+      if (!(type.prototype instanceof Transform) || !this.transform) {
+        this.addComponent(type);
+      }
+    }
```
Comment (singlecoder) [inline packages/core/src/Entity.ts:233]:
```typescript
for (let i = 0, n = components.length; i < n; i++) {
  const type = components[i];
  if (type.prototype instanceof Transform) {
    this._transform || this.addComponent(type);
  } else {
    this.addComponent(type);
  }
}
this._transform || this.addComponent(Transform);
```

This way, may be it's easier to understand and performs better.

File: packages/core/src/Entity.ts [hunk 3]
```diff
@@ -214,6 +249,11 @@ export class Entity extends EngineObject {
     const component = new type(this, ...args) as InstanceType<T>;
     this._components.push(component);
     component._setActive(true, ActiveChangeFlag.All);
+    if (component instanceof Transform) {
+      const preTransform = this._transform;
+      this.transform = component;
+      preTransform?.destroy();
```
Comment (singlecoder) [inline packages/core/src/Entity.ts:255]: Should remove component

File: packages/core/src/Entity.ts (hunk 2 above)
Comment (cptbtptpbcptdtptp) [reply packages/core/src/Entity.ts:225]: Done.

File: tests/src/core/Transform.test.ts (hunk 1 above)
Comment (cptbtptpbcptdtptp) [reply tests/src/core/Transform.test.ts:127]: Done.

**Review by GuoLei1990 — APPROVED** [review]

**Pull request merged**